daml-script/runner/src/main/scala/com/digitalasset/daml/lf/engine/script/v2/ledgerinteraction/IdeLedgerClient.scala

```diff
@@ -283,6 +283,90 @@ class IdeLedgerClient(
+      case UnhandledException(ty, v) =>
+        SubmitError.UnhandledException(getTypeIdentifier(ty).map(tyId => (tyId, v)))
+      case UserError(msg) => SubmitError.UserError(msg)
+      case _: TemplatePreconditionViolated => SubmitError.TemplatePreconditionViolated()
+      case CreateEmptyContractKeyMaintainers(tid, arg, _) =>
+        SubmitError.CreateEmptyContractKeyMaintainers(tid, arg)
+      case FetchEmptyContractKeyMaintainers(tid, keyValue) =>
+        SubmitError.FetchEmptyContractKeyMaintainers(GlobalKey.assertBuild(tid, keyValue))
+      case WronglyTypedContract(cid, exp, act) => SubmitError.WronglyTypedContract(cid, exp, act)
+      case ContractDoesNotImplementInterface(iid, cid, tid) =>
+        SubmitError.ContractDoesNotImplementInterface(cid, tid, iid)
+      case ContractDoesNotImplementRequiringInterface(requiringIid, requiredIid, cid, tid) =>
+        SubmitError.ContractDoesNotImplementRequiringInterface(cid, tid, requiredIid, requiringIid)
+      case NonComparableValues => SubmitError.NonComparableValues()
+      case ContractIdInContractKey(_) => SubmitError.ContractIdInContractKey()
+      case ContractIdComparability(cid) => SubmitError.ContractIdComparability(cid.toString)
+      case e @ Dev(_, innerError) =>
+        SubmitError.DevError(
+          innerError.getClass.getSimpleName,
+          Pretty.prettyDamlException(e).renderWideStream.mkString,
+        )
+    }
+  }
+
+  // Projects the scenario submission error down to the script submission error
+  private def fromScenarioError(err: scenario.Error): SubmitError = err match {
+    case scenario.Error.RunnerException(e: SError.SErrorCrash) =>
+      SubmitError.UnknownError(e.toString)
+    case scenario.Error.RunnerException(SError.SErrorDamlException(err)) =>
+      fromInterpretationError(err)
+
+    case scenario.Error.Internal(reason) => SubmitError.UnknownError(reason)
+    case scenario.Error.Timeout(timeout) => SubmitError.UnknownError("Timeout: " + timeout)
+
+    // We treat ineffective contracts (ie, ones that don't exist yet) as being not found
+    case scenario.Error.ContractNotEffective(coid, _, _) =>
+      SubmitError.ContractNotFound(coid)
+
+    case scenario.Error.ContractNotActive(coid, templateId, _) =>
+      SubmitError.ContractNotActive(templateId, coid)
+
+    // Similarly, we treat contracts that we can't see as not being found
+    case scenario.Error.ContractNotVisible(coid, _, _, _, _) =>
+      SubmitError.ContractNotFound(coid)
+
+    case scenario.Error.ContractKeyNotVisible(_, key, _, _, _) =>
+      SubmitError.ContractKeyNotFound(key)
+
+    case scenario.Error.CommitError(
+          ScenarioLedger.CommitError.UniqueKeyViolation(ScenarioLedger.UniqueKeyViolation(gk))
+        ) =>
+      SubmitError.DuplicateContractKey(gk)
+
+    case scenario.Error.LookupError(err, _, _) =>
+      // TODO[SW]: Implement proper Lookup error throughout
+      SubmitError.UnknownError("Lookup error: " + err.toString)
```

Comment:
Another ticket, following the package vetting logic, we should have a proper error for non-existent template (or any lookup failure). I'm not sure what the ledger-api does currently if you try to use a template that doesn't exist